documentation/processing.tex

```diff
@@ -26,16 +22,60 @@ \subsection{Overview}
 	\item \emph{Processing tracer} for tracing processed statements (see \cref{macro_tracer}).
 \end{itemize}
 
-\subsectionSrc{Types of statement processors}{parser\_library/src/processing/statement\_processors/statement\_processor.h}
-\label{lab06:sect_proc}
 
+\subsection{Composition}
+
+As the processing of the HLASM source file is rather complicated, we define a complex set of abstraction objects over the Gordian assembling of HLASM language:
+
+\begin{description}
+	\item[Statement provider] (see \cref{lab06:sect_prov}) is able to produce \TT{statement} structures. Its functionality is to provide statements from its finite statement stream.
+	
+	\item[Statement processor] (see \cref{lab06:sect_proc}) is an object that takes statement structures from a provider. Then, it performs a specific action with the acquired statement; namely, stores it into macro/copy definition (\emph{Macro/Copy processor}) or looks for sequence symbol (\emph{Lookahed processor}) or performs contained instruction (\emph{Opencode processor}).
```

Comment:
co nieco taketo? 
Then, it performs a specific action with the acquired statement; namely -> Then, depending on current context (whether the analyzer is processing macro, opencode or is in lookahead mode the statement processor does one of following:

Reply:
ja si myslim ze v tychto descriptions nechcem este viac zlozitit. tamto sa dozvie v danej sekcii. a nie je to uplne pravda, lebo statement procesor sa nerozhoduje podla kontextu, to manager sa rozhoduje ktory procesor nasadi, podla kontextu